File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        d = {}

        for x in nums:
            d[x] = d.get(x, 0) + 1

        for x in d:
            if d[x] > 1:
                return True
        return False

File: Data Structures & Algorithms/majority-element/submission-1.js
class Solution {
    majorityElement(nums) {
        let hash = {};
        for (let i=0; i<nums.length; i++) {
            if (hash[nums[i]]) {
                hash[nums[i]] += 1;
            } else {
                hash[nums[i]] = 1;
            }

            if (hash[nums[i]] > nums.length/2) {
                return nums[i];
            }
        }
    }
}
